src/plugins/intel_gpu/src/graph/impls/ocl/non_zero.cpp

```diff
@@ -48,20 +54,26 @@ struct gather_nonzero_impl : typed_primitive_impl_ocl<gather_nonzero> {
         return make_unique<gather_nonzero_impl>(*this);
     }
 
+public:
     static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param) {
         auto params = get_default_params<kernel_selector::gather_nonzero_params>(impl_param);
         auto optional_params = get_default_optional_params<kernel_selector::gather_nonzero_optional_params>(impl_param.get_program());
 
         params.inputs.push_back(convert_data_tensor(impl_param.get_input_layout(1)));
-        params.ov_input_rank = impl_param.get_input_layout().get_shape().size();
+        params.ov_input_rank = impl_param.get_input_layout().get_partial_shape().rank().get_length();
```

Comment:
`get_partial_shape().size()` can be used instead

Reply:
updated. thanks